tests/main/services-user/task.yaml

```diff
@@ -0,0 +1,98 @@
+
+details: |
+    With the addition of --user, --users and --system switches for snap operations like
+    snap start/stop/restart, we want to test that we can select user and/or system services,
+    when performing these operations.
+
+# Only run on systems with polkit rules are supported
+systems: [ ubuntu-2* ]
+
+prepare: |
+    snap set system experimental.user-daemons=true
+    tests.session kill-leaked
+    tests.session -u test prepare
+
+    cat <<\EOF >/etc/polkit-1/localauthority/50-local.d/spread.pkla
+    [Normal Staff Permissions]
+    Identity=unix-user:test
+    Action=io.snapcraft.snapd.manage
+    ResultAny=yes
+    ResultInactive=no
+    ResultActive=yes
+    EOF
+
+restore: |
+    tests.session -u test restore
+    snap unset system experimental.user-daemons
+
+debug: |
+    tests.session dump
+    tests.session -u test exec systemctl --user status snapd.session-agent.service || true
+    tests.session -u test exec journalctl --user || true
+
+execute: |
+    echo "Installing the service snap"
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-user-service
+
+    echo "We can see the system services running"
+    systemctl status snap.test-snapd-user-service.svc3.service | MATCH "running"
+    systemctl status snap.test-snapd-user-service.svc4.service | MATCH "running"
+
+    echo "We can see the user services running"
+    tests.session -u test exec systemctl --user is-active snap.test-snapd-user-service.svc1.service | MATCH "active"
+    tests.session -u test exec systemctl --user is-active snap.test-snapd-user-service.svc2.service | MATCH "active"
+
+    # if root is making this request, implied scopes are all
+    echo "(root) Stopping all services for snap"
+    snap stop test-snapd-user-service
+    systemctl status snap.test-snapd-user-service.svc3.service | MATCH "inactive"
+    systemctl status snap.test-snapd-user-service.svc4.service | MATCH "inactive"
+    tests.session -u test exec systemctl --user is-active snap.test-snapd-user-service.svc1.service | MATCH "inactive"
+    tests.session -u test exec systemctl --user is-active snap.test-snapd-user-service.svc2.service | MATCH "inactive"
+
+    # if non-root are making requests, and user services are the target,
+    # then its an error to not specify --user scope
+    echo "(user) Trying to start services for snap, which will fail with no scope set"
+    tests.session -u test exec snap start test-snapd-user-service 2>&1 | tr '\n' ' ' | tr -s ' ' | MATCH "non-root users must specify service scope when targeting user services"
+    
+    echo "(user) Starting user services for snap with scope --user"
+    tests.session -u test exec snap start --user test-snapd-user-service
+
+    echo "And we can see the user services running"
+    tests.session -u test exec systemctl --user is-active snap.test-snapd-user-service.svc1.service | MATCH "active"
+    tests.session -u test exec systemctl --user is-active snap.test-snapd-user-service.svc2.service | MATCH "active"
```

Comment:
it'd be nice to have another user here as well, but I don't remember if we add more users than just `test`

Reply:
good point, let me just check on this and see if I can add one extra

Reply:
added an additional user for testing in the spread test

Reply:
thank you!